elasticsearch/connection/http_urllib3.py

```diff
@@ -141,10 +141,10 @@ def perform_request(self, method, url, params=None, body=None, timeout=None, ign
             if not isinstance(method, str):
                 method = method.encode('utf-8')
 
+            request_headers = dict(self.headers)
             if headers:
```

Comment:
The `if` block seems redundant now, since the condition that custom headers have been passed is applied in the `update` function call. What do you think?

Reply:
Good point. I pushed another commit to address this.